### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by BenjaminParment**
I can add some test after im sure I did this correctly

### CODE REVIEW

File: src/shared/network/NetworkService.ts [hunk 1]
```diff
@@ -27,6 +35,20 @@ export class NetworkService {
...
+        let message: string | undefined;
+        switch (error.status) {
+            case 401:
+                state.dispatch(setLoggedOut());
+            case 500:
+                message = error.message;
+            default:
+                message = getDefaultMessage(error.status);
+        }
+        message && state.dispatch(setError(message));
```
Comment (j-n-white) [inline src/shared/network/NetworkService.ts]: If message is undefined then we still want to display a generic message

File: src/shared/utils/networkUtils.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const getDefaultMessage = (status: number): string => {
+    switch (status) {
+        case 400:
+            return "Malformed request syntax";
```
Comment (j-n-white) [inline src/shared/utils/networkUtils.ts]:
```suggestion
            return "Invalid request";

```

File: src/shared/utils/networkUtils.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export const getDefaultMessage = (status: number): string => {
+    switch (status) {
+        case 400:
+            return "Malformed request syntax";
+        case 403:
+            return "Unauthorized access";
```
Comment (j-n-white) [inline src/shared/utils/networkUtils.ts]:
```suggestion
            return "You are not authorised to view this resource";

```

File: src/shared/utils/networkUtils.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export const getDefaultMessage = (status: number): string => {
+    switch (status) {
+        case 400:
+            return "Malformed request syntax";
+        case 403:
+            return "Unauthorized access";
+        case 404:
+            return "Request not found";
```
Comment (j-n-white) [inline src/shared/utils/networkUtils.ts]:
```suggestion
            return "Resource not found";

```

File: src/shared/utils/networkUtils.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const getDefaultMessage = (status: number): string => {
+    switch (status) {
+        case 400:
+            return "Malformed request syntax";
+        case 403:
+            return "Unauthorized access";
+        case 404:
+            return "Request not found";
+        case 405:
+            return "Method not allowed";
```
Comment (j-n-white) [inline src/shared/utils/networkUtils.ts]: I don't see this is worth a specific message

File: src/shared/utils/networkUtils.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        case 400:
+            return "Malformed request syntax";
+        case 403:
+            return "Unauthorized access";
+        case 404:
+            return "Request not found";
+        case 405:
+            return "Method not allowed";
+        default:
+            return "Please reload";
```
Comment (j-n-white) [inline src/shared/utils/networkUtils.ts]:
```suggestion
            return "Something went wrong, please reload";

```

File: src/components/submissions/state/submissionsActions.ts [hunk 7]
```diff
@@ -8,7 +8,7 @@ import { ThunkResult } from "state/rootReducer";
 import { constructSubmissionFilterQuery } from "../submissions-filter/submissionsFilterMapper";
 
 const submissionsUrl = "/submissions";
-const sbmissionsUrlQuery = "/submissions/search";
+const sbmissionsUrlQuery = "/submissionsss/search";
```
Comment (j-n-white) [inline src/components/submissions/state/submissionsActions.ts]: wat?

File: src/shared/network/NetworkService.ts [hunk 8]
```diff
@@ -63,11 +75,11 @@ export class NetworkService {
                 }),
             })
             .then(response => response.data)
-            .catch(err => {
+            .catch((err: ErrorObject) => {
                 if (axios.isCancel(err)) {
                     return undefined;
                 }
-                return this.handleError(err);
+                this.handleError(err);
```
Comment (j-n-white) [inline src/shared/network/NetworkService.ts]: We probably still want to return that promise

File: src/components/submissions/state/submissionsActions.ts (hunk 7 above)
Comment (BenjaminParment) [reply src/components/submissions/state/submissionsActions.ts]: shoot, my bad, I was checking what happens when we hit a bad endpoint..

**Pull request merged**